api/api.py

```diff
@@ -294,3 +294,65 @@ def custom_build(ref: str, command: str, deps: List[str], tag: str = "", disable
     disable_push: whether Tilt should push the image in to the registry that the Kubernetes cluster has access to. Set this to true if your command handles pushing as well.
   """
   pass
+
+class LiveUpdateStep:
+    """A step in the process of performing a LiveUpdate on an image's container.
+
+    See :meth:`live_update`.
+    """
+    pass
+
+def sync(local_path: str, remote_path: str) -> LiveUpdateStep:
+    """Specify that any changes to `localPath` should be synced to `remotePath`
+
+    May not follow any `run` steps in a `live_update`.
+
+    Args:
+        localPath: A path relative to the Tiltfile's directory. Changes to files matching this path will be synced to `remotePath`.
+            Can be a file (in which case just that file will be synced) or directory (in which case any files recursively under that directory will be synced).
+        remotePath: container path to which changes will be synced. Must be absolute.
+    """
+    pass
+
+def run(cmd: str, trigger: Union[List[str], str] = []) -> LiveUpdateStep:
+    """Specify that the given `cmd` should be executed when updating an image's container
+
+    May not precede any `sync` steps in a `live_update`.
+
+    Args:
+      cmd: A shell command.
+      trigger: If the ``trigger`` argument is specified, the build step is only run on changes to the given file(s).
+    """
+    pass
+
+def container_restart() -> LiveUpdateStep:
+    """Specify that a container should be restarted when it is live-updated.
+
+    May only be included in a `live_update` once, and only as the last step.
+
+    """
+    pass
+
+def live_update(image: str, steps: List[LiveUpdateStep], full_rebuild_trigger: Union[str, List[str]] = []) -> None:
+    """Specify how to update containers running `image`.
+
+    The list of steps must be, in order:
+
+    - 0 or more :meth:`sync`
```

Comment:
meth? :eyes:

Reply:
this makes `sync` a hyperlink to the method `sync`, instead of just monospaced text

https://stackoverflow.com/a/22714510

Reply:
:breaking-bad-this-is-not-meth-dot-gif: